pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -74,27 +73,12 @@ func (f *Function) validate(fnType string, idx int) error {
 				Reason: err.Error(),
 			}
 		}
-		configFields = append(configFields, "configPath")
-	}
-	if len(f.ConfigMap) != 0 {
-		configFields = append(configFields, "configMap")
-	}
-	if !IsNodeZero(&f.Config) {
-		config := yaml.NewRNode(&f.Config)
-		if _, err := config.GetMeta(); err != nil {
+		if len(f.ConfigMap) != 0 {
 			return &ValidateError{
-				Field:  fmt.Sprintf("pipeline.%s[%d].config", fnType, idx),
-				Reason: "functionConfig must be a valid KRM resource with `apiVersion` and `kind` fields",
+				Field: fmt.Sprintf("pipeline.%s[%d]", fnType, idx),
+				Reason: "only one of 'configMap', 'configPath' can be specified; got both",
```

Comment:
An alternative messaging that is more inline with http://go/kpt-errors-style-guide

functionConfig must not specify `configMap` and `configPath` both at the same time.